daemon/src/address_map.rs

```diff
@@ -61,6 +61,32 @@ where
             _ => Err(NotConnected::new::<A>()),
         }
     }
+
+    /// Sends a message to the actor stored with the given key.
+    pub async fn send_fallible<M>(
+        &self,
+        key: &K,
+        msg: M,
+    ) -> Result<anyhow::Result<()>, NotConnected>
+    where
+        M: Message<Result = anyhow::Result<()>>,
+        A: Handler<M> + ActorName,
+    {
+        match self.inner.get(key) {
+            Some(addr) if addr.is_connected() => {
+                if let Err(e) = addr
+                    .send(msg)
+                    .await
+                    .expect("we checked that we are connected")
+                {
+                    return Ok(Err(e));
+                };
+
+                Ok(Ok(()))
```

Comment:
Isn't this equivalent:

```suggestion
                let result = addr.send(msg).await.expect("we checked that we are connected");
                Ok(result)
```

Reply:
Equivalent and better.